framework/src/application/node/processor/processor.js

```diff
@@ -335,7 +335,8 @@ class Processor {
 	}
 
 	async _deleteBlock(block, processor, saveTempBlock = false) {
-		const stateStore = await this.chainModule.newStateStore();
+		// Offset must be set to 1, because lastBlock is still this deleting block
+		const stateStore = await this.chainModule.newStateStore(1);
```

Comment:
With this usage, it meant that you want to skip the `lastBlock` in `stateStore` cache. But its confusing, because `offset` term is used usually to offset from beginning.  So while reading,

```
await this.chainModule.newStateStore({heightOffset: 5})
```

I would assume that you want to skip the first five block and want to start stateStore from heigh 6. 

We should use some other term based on usage. May be `skipLast`, then it would be more readable. 

```
await this.chainModule.newStateStore({skipLast: 5})
```

Reply:
i will rename the argument to be `skipLastHeights`?